Sources/LanguageServerProtocol/SupportTypes/RegistrationOptions.swift

```diff
@@ -26,21 +37,57 @@ public struct TextDocumentRegistrationOptions: RegistrationOptions {
     self.documentSelector = documentSelector
   }
 
-  public init?(fromLSPDictionary dictionary: [String : LSPAny]) {
-    guard let selectorValue = dictionary[CodingKeys.documentSelector.stringValue] else {
-      self.documentSelector = nil
-      return
+  public func encodeIntoLSPAny(dict: inout [String: LSPAny]) {
+    guard let documentSelector = documentSelector else { return }
+    dict["documentSelector"] = documentSelector.encodeToLSPAny()
+  }
+
+  public static func == (lhs: TextDocumentRegistrationOptions, rhs: TextDocumentRegistrationOptions) -> Bool {
```

Comment:
You should be able to use the default synthesized implementations for `==` and `hash(into:)` now, right?

Reply:
Done